code/ui/generalhud/scoreboard/Scoreboard.cs

```diff
@@ -70,13 +70,11 @@ private void OnPlayerSpawned(TTTPlayer player)
         private void AddPlayer(PlayerScore.Entry entry)
         {
             ScoreboardGroup scoreboardGroup = GetScoreboardGroup(entry);
-            ScoreboardEntry scoreboardEntry = scoreboardGroup.AddEntry(entry);
-
             scoreboardGroup.GroupMembers++;
+            scoreboardGroup.UpdateLabel();
 
-            _entries.Add(entry.Id, scoreboardEntry);
+            _entries.Add(entry.Id, scoreboardGroup.AddEntry(entry));
```

Comment:
Just reordering for readability, I did some testing and it seemed like it had no issues.